pkg/policy/api/rule_validation.go

```diff
@@ -199,3 +213,47 @@ func (cidr CIDR) Validate() error {
 
 	return nil
 }
+
+// Validate validates a CIDRRule by checking that the CIDR itself is valid,
+// and ensuring that all of the exception CIDRs are contained within the CIDR.
+func (c CIDRRule) Validate() error {
+	err := c.Cidr.Validate()
+	if err != nil {
+		return err
+	}
+
+	// Don't need to check error as this was done in validation of CIDR.
+	_, cidrNet, err := net.ParseCIDR(string(c.Cidr))
+
+	if err == nil {
+		// Returns the prefix length as zero if the mask is not continuous.
+		ones, _ := cidrNet.Mask.Size()
+		if ones == 0 {
+			return fmt.Errorf("Mask length can not be zero")
+		}
+
+		// Ensure that each provided exception CIDR is contained within the CIDR
+		// to/from which we want to allow traffic.
+		for _, p := range c.ExceptCIDR {
+			exceptCIDRAddr, _, err := net.ParseCIDR(string(p))
+			if err != nil {
+				return err
+			}
+			if !cidrNet.Contains(exceptCIDRAddr) {
+				return fmt.Errorf("allow CIDR %s does not contain exclude CIDR %s", c.Cidr, p)
+			}
+		}
+	} else {
+		// Try to parse as a fully masked IP or an IP subnetwork
+		ip := net.ParseIP(string(c.Cidr))
+		if ip == nil {
+			return fmt.Errorf("Unable to parse CIDR: %s", err)
+		}
+
+		if len(c.ExceptCIDR) > 0 {
+			return fmt.Errorf("IP was provided, cannot exclude a CIDR from an IP")
```

Comment:
IP -> IP address
CIDR -> CIDR prefix

Reply:
I decided to remove this, and ensure that only strings that follow the proper CIDR notation (<IP>/<prefix>) are allowed, as this is what Kubernetes does. If a user wants to supply a single IP, they can provide it as a `/32` or a `/128`.